docs/tutorials/how-tos/testing-devnet-features.md

```diff
@@ -0,0 +1,154 @@
+---
+html: testing-prerelease-devnet-features.html
+parent: how-tos.html
+seo:
+    description: Guide on using pre-release transaction types on the XRP Ledger for developers.
+labels:
+  - Development
+  - Customization
+  - Devnet
+---
+# Test a Pre-Release Transaction Type on the XRP Ledger
+
+_(Requires cloning and modifying XRPL core repositories and understanding of XRPL transaction serialization)_
+
+Pre-release transactions are [Amendments](../../concepts/networks-and-servers/amendments.md) that represent new features or other changes to transaction processing. Features are typically released to the [XRPL Devnet](../../concepts/networks-and-servers/parallel-networks.md) for early testing.
+
+This guide walks through using either JavaScript with `xrpl.js` or Python with `xrpl-py`. This approach is typically only necessary for early pre-release amendments on the [XRPL Devnet](../../concepts/networks-and-servers/parallel-networks.md).
+
+**Note**: The code samples below illustrate how to prepare your development environment and modify the XRPL to support custom transaction types, using the respective [client library](../../references/client-libraries.md).
+
+## Prerequisites
+
+- Basic understanding of XRPL transactions.
+- Development environment setup for JavaScript or Python.
+
+## Steps
+
+### 1. Set Up Your Development Environment
+
+#### JavaScript:
+Ensure Node.js and npm are installed. Use npm to install `xrpl.js` in your project:
+```javascript
+npm install xrpl
+```
+
+#### Python:
+Ensure Python and pip are installed. Use pip to install `xrpl-py`:
+```bash
+pip install xrpl-py
+```
+
+### 2. Clone XRPL Core Repositories
+
+Clone the `rippled` source code and the `xrpl-codec-gen` tool from GitHub:
+```bash
+git clone https://github.com/XRPLF/rippled
+git clone https://github.com/RichardAH/xrpl-codec-gen
+```
+
+### 3. Generate Definitions File
+
+Ensure you're on the `develop` branch of `rippled` to get the latest changes. The `develop` branch is kept in sync with the [XRPL Devnet](../../concepts/networks-and-servers/parallel-networks.md).
+
+Use `xrpl-codec-gen` to create a `definitions.json` file based on the XRPL source code. There is no difference between using Python or JavaScript to generate the `definitions.json`.
```

Comment:
Right, you mean run the server command? We have a PR to do custom definitions, but it wasn't finished yet.

Reply:
I am saying you can use that result and paste it but not have to generate from source

Reply:
@ckniffen Do you know of a utility that does this as a cli? I agree that is the better way, but this is a way that someone can just run a command and get the file output, versus needing to make their own util to do so.

I think we should probably have that util be updated, so that it can be done via CLI and pass in a rippled url.

Reply:
You dont need a utility.  Just start up a rippled server or point to an existing one and get the result of this command. Doesn't require downloading source and another utility. https://xrpl.org/docs/references/http-websocket-apis/public-api-methods/server-info-methods/server_definitions/

Reply:
Updated to use `docker` and `curl` instead